sql/log.cc

```diff
@@ -1476,7 +1476,7 @@ void Query_logger::cleanup()
 bool Query_logger::slow_log_write(THD *thd, const char *query,
                                   size_t query_length)
 {
-  DBUG_ASSERT(thd->enable_slow_log && opt_slow_log);
+  DBUG_ASSERT(thd->enable_slow_log);
```

Comment:
Looks good. Just one question: When is this set to false?

Reply:
this == thd->enable_slow_log

Reply:
Main site is THD::THD.